osc-server/src/test/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/CreatePortGroupTaskTest.java

```diff
@@ -0,0 +1,302 @@
+			port = new VMPort((Subnet) protectionEntity, "mac-address", UUID.randomUUID().toString(),
+					UUID.randomUUID().toString(), null);
+		}
+
+		this.vmProtectedPorts.add(port);
+		return newSGM(protectionEntity, sgmId);
+	}
+
+	private SecurityGroupMember newSGM(OsProtectionEntity protectionEntity, Long sgmId) {
+		// TODO emanoel: Remove this mock once the SGM is no longer kept in a
+		// TreeSet in the SGM.
+		SecurityGroupMember sgm = Mockito.spy(new SecurityGroupMember(protectionEntity));
+		Mockito.doReturn(-1).when(sgm).compareTo(Mockito.any());
+		sgm.setId(sgmId);
+
+		return sgm;
+	}
+
+	private VMPort newVMPort(VM vm) {
+		return new VMPort(vm, "mac-address" + UUID.randomUUID().toString(), UUID.randomUUID().toString(),
+				UUID.randomUUID().toString(), null);
+	}
+
+	private SecurityGroup createSecurityGroup(Long vcId, String tenantId, String tenantName, Long sgId, String sgName) {
+		VirtualizationConnector vc = new VirtualizationConnector();
+		vc.setId(1L);
+		SecurityGroup sg = new SecurityGroup(vc, tenantId, tenantName);
+		sg.setId(sgId);
+		sg.setName(sgName);
+
+		when(this.em.find(SecurityGroup.class, sg.getId())).thenReturn(sg);
+		return sg;
+	}
+
+	private List<NetworkElement> ostRegisterPorts(SecurityGroup sg) throws Exception {
+		List<NetworkElement> neList = this.vmProtectedPorts.stream().map(NetworkElementImpl::new)
+				.collect(Collectors.toList());
+		PowerMockito.doReturn(neList).when(OpenstackUtil.class, "getPorts",
+				sg.getSecurityGroupMembers().iterator().next());
+		return neList;
+	}
+
+	private void registerNetworkRedirectionApi(SdnRedirectionApi redirectionApi, VirtualizationConnector vc)
+			throws Exception {
+		PowerMockito.spy(SdnControllerApiFactory.class);
+		PowerMockito.doReturn(redirectionApi).when(SdnControllerApiFactory.class, "createNetworkRedirectionApi", vc);
+	}
+
+	private SdnRedirectionApi mockRegisterNetworkElement(NetworkElement ne) throws Exception {
+		SdnRedirectionApi redirectionApi = mock(SdnRedirectionApi.class);
+		when(redirectionApi.registerNetworkElement(networkElementCaptor.capture())).thenReturn(ne);
+
+		return redirectionApi;
+	}
+
+	private void registerDomain(String domainId, SecurityGroup sg) throws Exception {
+		PowerMockito.doReturn(domainId).when(OpenstackUtil.class, "extractDomainId", eq(sg.getTenantId()),
+				eq(sg.getTenantName()), eq(sg.getVirtualizationConnector()), domainIdNetworkElementCaptor.capture());
+	}
+
+	private NetworkElement createNetworkElement() throws Exception {
+		VMPort port = null;
+		OsProtectionEntity protectionEntity;
+		protectionEntity = new VM("region", UUID.randomUUID().toString(), "name");
+		port = newVMPort((VM) protectionEntity);
+		NetworkElementImpl ne = new NetworkElementImpl(port);
+
+		return ne;
+	}
+
+	private void assertNetworkElementsParentIdWithDomainId(List<NetworkElement> neList, String domainId) {
+		for (NetworkElement ne : neList) {
+			Assert.assertEquals(ne.getParentId(), domainId);
```

Comment:
Ensure to have a message on all your asserts (see unit tests guidelines), i.e.: "Network element parent id mismatch". This will place the message in the failure making it clearer.